src/getters/crunchbase.py

```diff
@@ -0,0 +1,114 @@
+"""
+discovery_utils.src.getters.crunchbase.py
+
+Module for easy access to downloaded CB data on S3.
+
+"""
+import pandas as pd
+import boto3
+import logging
+import re
+
+
+# Define global variables
+BUCKET_NAME = "discovery-iss"
+S3_PATH = "data/crunchbase/"
+AWS_ACCESS_KEY =
+AWS_SECRET_KEY = 
+FILE_NAMES = [
+    "acquisitions.csv",
+    "category_groups.csv",
+    "checksum.csv",
+    "degrees.csv",
+    "event_appearances.csv",
+    "events.csv",
+    "funding_rounds.csv",
+    "funds.csv",
+    "investment_partners.csv",
+    "investments.csv",
+    "investors.csv",
+    "ipos.csv",
+    "jobs.csv",
+    "org_parents.csv",
+    "organization_descriptions.csv",
+    "organizations.csv",
+    "people_descriptions.csv",
+    "people.csv"
+]
+
+
+# Initialize S3 client
+S3 = boto3.client(
+    "s3", aws_access_key_id=AWS_ACCESS_KEY, aws_secret_access_key=AWS_SECRET_KEY
+    )
+
+
+# Building block functions - not to be used directly
+
+def list_s3_directories() -> list[str]:
+    """List S3 directories under the specified prefix."""
+    # List all objects with the specified prefix
+    objects = S3.list_objects_v2(Bucket=BUCKET_NAME, Prefix=S3_PATH, Delimiter='/')
+
+    # Extract the directory names from the CommonPrefixes list
+    directories = [prefix['Prefix'] for prefix in objects.get('CommonPrefixes', [])]
+
+    return directories
+
+def extract_timestamp(directory: str) -> str or None:
+    """ Use regular expression to extract the timestamp from a directory name"""
+    match = re.search(r'\d{8}_\d{6}', directory)
+    if match:
+        timestamp_str = match.group()
+        return timestamp_str
+    else:
+        return None
+    
+def directory(position: int) -> str:
+    """Directory name for a given snapshot of CB data."""
+    
+    # Get the list of directories
+    directories = list_s3_directories()
+    
+    # Get a list of timestamps
+    timestamps = []
+    for directory in directories:
+        timestamps.append(extract_timestamp(directory))
+    
+    # Sort the timestamps in descending order
+    sorted_timestamps = sorted(timestamps, reverse=True)
```

Comment:
You could store sorted and timestamp extracted `directories` globally for this module (i.e. CRUNCHBASE_DIRS), then in `latest_table` just refer to CRUNCHBASE_DIRS[0]

Reply:
Thanks @sqr00t - is this so that you make the utils more simple by factoring out the list generation code?

But the difference with these global variables is that they would need to be created with other utils for each workflow run, right? Variables like `BUCKET_NAME` and `AWS_ACCESS_KEY` are constant.

Reply:
In the context of getters in this module, will the directories list sorted by timestamp change whether it's `latest` or `secondlatest`?

Reply:
Nope, I just meant if you ran the workflow a week later, you'd have a new list. @sqr00t

Reply:
So in a week later, when you run this workflow twice or more on the same day, will the directories list sorted by timestamp change?

Or even if today, you ran this workflow multiple times, will the directories list sorted by timestamp change within today?

Reply:
I guess it depends if crunchbase_to_s3 is included in the workflow. Because this is the script which pulls data and creates a versioned snapshot in S3. @sqr00t